cms_coding_rules.md

```diff
@@ -0,0 +1,144 @@
+28. Try to avoid excessively long lines of code that impair readability.
+29. Data members of a class must not be redeﬁned in derived classes since doing so hides the original data member and could create confusion between the original and redefined data members.
+
+## 5 -- Documentation Rules
+1. Always  comment  complex,  tricky,  or  non-intuitive  portions of  code.
+2. When revising code, be sure to update and revise comments.
+
+## 6 -- Packaging Rules
+#### Libraries
+1. It is discouraged to have small packages (with couple of cc and header files).
+2. Functionality used by multiple packages should go in the `.../src` and `.../interface` directories.
+3. Only header files that expose a public interface should go into the `.../interface` directory.  
+4. Include only files that are in the current directory, e.g.  
+`#include "some_header.h"`  
+ or in the `.../interface directory`, e.g  
+`#include "Subsystem/Package/interface/some_header.h"`
+#### Plugins
+5. Put plugins (e.g. EDProducers, EDAnalyzers, etc.) into a `Package/Subpackage/plugins/` directory, with its dedicated `BuildFile.xml`.
+6. Do not split plugins into header and source files. If you do split them, keep the header files in the `.../plugins` directory.
+7. All code used only by the plugins in `../plugins` should also go under `.../plugins`.
+ `plugins.cc` and `SealPlugins.cc` or any special files that just define plugins, except for template instantiations, are discouraged. 
+#### Tests
+8. Unit tests to test the functionalities of your Library and/or Plugins should go under `.../test`.
+9. Add test library/plugins in `.../test/BuildFile.xml` for the common functionality used only by your unit tests.
+10. For unit tests which simply run `cmsRun your-cfg` (to test your plugin), please use `<test name="..." command="cmsRun …"/>` in your `.../test/BuildFile.xml`.
+#### Python
+11. A `_cfi` file should contain only the definition of one module, and possible Modifier ("era") customizations on it. The module label should be the same as the `_cfi` file name.
+12. The `_cfi` file should be left to be generated automatically with the `fillDescriptions()`. When Modifier customizations are needed, the auto-generated label should have e.g. "Default" postfix and be imported+cloned to the desired name.
+13. A module/Task/Sequence/Path with a given name should be defined in exactly one `_cfi` or `_cff` file.
+14. All Modifier customizations on a module/Task/Sequence/Path should be applied on the same file that defines the module/Task/Sequence/Path.
+15. When one customizes an existing parameter in `clone()`, `Modifier.toModify()`, or in assignment, explicit types on the right hand side should be avoided.
+#### Data files
+16. To keep the repository size under control, we discourage adding any data files under `.../data`.
+17. Please use cms-data externals github repositories to add/change any data file.
```

Comment:
add a link to cms-data org on GitHub

Reply:
yes please, use https://github.com/cms-data/

Reply:
Added link.